Sources/CacheAdvance/CacheAdvance.swift

```diff
@@ -130,10 +121,13 @@ public final class CacheAdvance<T: Codable> {
         try setUpFileHandlesIfNecessary()
 
         var messages = [T]()
-        while let encodedMessage = try reader.nextEncodedMessage(cacheOverwritesOldMessages: shouldOverwriteOldMessages) {
+        while let encodedMessage = try reader.nextEncodedMessage() {
             messages.append(try decoder.decode(T.self, from: encodedMessage))
         }
 
+        // Now that we've read all messages, seek back to the oldest message.
+        try reader.seekToBeginningOfOldestMessage()
```

Comment:
I should have commented on this earlier: I moved this code from `nextEncodedMessage()`. It was doing this seek at the end, which felt like the wrong separation of responsibilities. So no test caught this issue, since it was never an issue.

That said, having the high test coverage that we have here helped in numerous ways when creating this PR.

1. It helped me make pretty substantial changes with confidence. Removing the last remnants of `endOfNewestMessageMarker` only took 10 minutes. It would have taken way more to change + validate + fix any introduced issues if we didn't have as strong test coverage.
1. It helped me identify abandoned code. After making sweeping changes, seeing what lines of code didn't have coverage helped me identify code that could be simplified or no longer needed to exist. This was an unexpected benefit of having high coverage, and it rocked.

Reply:
👍 👍 
That dead code benefit is very cool. I hadn’t considered that benefit of tests either.